### ISSUE TITLE
Flutter WebView throws NullPointerException when trying to call onCreateInputConnection after state rebuilds

### ISSUE THREAD

**Issue by joelbrostrom** [flutter/flutter#36453]
## Steps to Reproduce

1.  Build `StatefulWidget` with a `WebView`
2. Navigate some web pages with text input fields (No problem)
3. Rebuild state with new widget (in my case I use BLoC with BlockBuilder)
4. Return to previous state and rebuild WebView (no problem)
5. Navigate web pages without text input field (no problem)
6. Navigate web page with text input field (crash)

## Code

```
Widget build(BuildContext context) {
    final HomeBloc _homeBloc = BlocProvider.of<HomeBloc>(context);

    return BlocBuilder<HomeEvent, HomeBlocState>(
      bloc: _homeBloc,
      builder: (BuildContext context, HomeBlocState state) {

        if (state is HomeWebview) {
          return WebView(
            initialUrl: _tabUrls[_currentTabIndex],
            navigationDelegate: navigationListener(_homeBloc),
            javascriptMode: JavascriptMode.unrestricted,
            onWebViewCreated: (webViewController) {
              _webViewController = webViewController;
            },
          );

       if (state is HomeLoginView) {
          return LoginPage();
       }
```

## Logs

```
W/System.err(30353): java.lang.NullPointerException: Attempt to invoke virtual method 'android.view.inputmethod.InputConnection android.view.View.onCreateInputConnection(android.view.inputmethod.EditorInfo)' on a null object reference
W/System.err(30353): 	at io.flutter.plugin.editing.TextInputPlugin.createInputConnection(TextInputPlugin.java:195)
W/System.err(30353): 	at io.flutter.view.FlutterView.onCreateInputConnection(FlutterView.java:397)
W/System.err(30353): 	at android.view.inputmethod.InputMethodManager.startInputInner(InputMethodManager.java:1763)
W/System.err(30353): 	at android.view.inputmethod.InputMethodManager.restartInput(InputMethodManager.java:1682)
W/System.err(30353): 	at czj.a(PG:7)
W/System.err(30353): 	at org.chromium.content.browser.input.ImeAdapterImpl.q(PG:289)
W/System.err(30353): 	at org.chromium.content.browser.input.ImeAdapterImpl.updateState(PG:190)
W/System.err(30353): 	at android.os.MessageQueue.nativePollOnce(Native Method)
W/System.err(30353): 	at android.os.MessageQueue.next(MessageQueue.java:326)
W/System.err(30353): 	at android.os.Looper.loop(Looper.java:181)
W/System.err(30353): 	at android.app.ActivityThread.main(ActivityThread.java:7045)
W/System.err(30353): 	at java.lang.reflect.Method.invoke(Native Method)
W/System.err(30353): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
W/System.err(30353): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:965)
F/chromium(30353): [FATAL:jni_android.cc(249)] Please include Java exception stack in crash report
F/libc    (30353): Fatal signal 5 (SIGTRAP), code 1 (TRAP_BRKPT), fault addr 0x7dee59b91c in tid 30353 (), pid 30353 ()
*** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
Build fingerprint: 'samsung/dreamltexx/dreamlte:9/PPR1.180610.011/G950FXXS5DSF1:user/release-keys'
Revision: '10'
ABI: 'arm64'
pid: 30353, tid: 30353, name:  >>> <<<
signal 5 (SIGTRAP), code 1 (TRAP_BRKPT), fault addr 0x7dee59b91c
Abort message: '[FATAL:jni_android.cc(249)] Please include Java exception stack in crash report
'
    x0  0000007fd5203fa8  x1  0000007dc02ce4c0  x2  0000000000000400  x3  0000007e83eb13c8
    x4  0000000000000021  x5  8080800000000080  x6  fefeff7d88c083ff  x7  7f7f7f7fffffff7f
    x8  0000000000000000  x9  0000000000000000  x10 0000007fd5203fa8  x11 00000000ffffff80
    x12 0000007dc02ce4c0  x13 ffffffffffffffff  x14 ffffffffff000000  x15 ffffffffffffffff
    x16 0000007e83eaf1b0  x17 0000007e83e45e68  x18 0000007fd52031ca  x19 0000007fd5204420
    x20 0000007fd5204428  x21 0000007fd5204430  x22 000000000000004f  x23 0000007e83eb13c8
    x24 0000007df0231000  x25 0000000000000000  x26 0000000000000000  x27 0000007e89c185f8
    x28 00000000ffffffff  x29 0000007fd5204410
    sp  0000007fd5203f60  lr  0000007dee59b7e0  pc  0000007dee59b91c
backtrace:
    #00 pc 0000000002a4291c  /data/app/com.android.chrome-zoKAZk1Jp81mwXfdxoly0A==/base.apk (offset 0x22c1000)
Lost connection to device.
***
```

### Flutter analyze
```
No issues found! (ran in 3.2s)
```

### Flutter Doctor -v
```
[✓] Flutter (Channel stable, v1.7.8+hotfix.3, on Mac OS X 10.14.5 18F132, locale en-SE)
    • Flutter version 1.7.8+hotfix.3 at /Users/joelbrostrom/Library/flutter
    • Framework revision b712a172f9 (9 days ago), 2019-07-09 13:14:38 -0700
    • Engine revision 54ad777fd2
    • Dart version 2.4.0

[!] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/joelbrostrom/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • ANDROID_HOME = /Users/joelbrostrom/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)
    ✗ Android license status unknown.
      Try re-installing or updating your Android SDK Manager.
      See https://developer.android.com/studio/#downloads or visit https://flutter.dev/setup/#android-setup for detailed instructions.

[✓] Xcode - develop for iOS and macOS (Xcode 10.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.2.1, Build version 10E1001
    • CocoaPods version 1.6.1

[✓] iOS tools - develop for iOS devices
    • ios-deploy 1.9.4

[✓] Android Studio (version 3.4)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 37.1.1
    • Dart plugin version 183.6270
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)

[✓] VS Code (version 1.36.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.2.0

[✓] Connected device (1 available)
    • SM G950F • 988a16355332374847 • android-arm64 • Android 9 (API 28)

! Doctor found issues in 1 category.

```

**Comment by shingohu** [comment]
the same issue

**Comment by GyuriMajercsik** [comment]
Found this reported by Crashlytics. It might be related. 

The last action logged is that the user tried to use the share functionality. 

Fatal Exception: java.lang.NullPointerException: Attempt to invoke virtual method 'android.view.inputmethod.InputConnection android.view.View.onCreateInputConnection(android.view.inputmethod.EditorInfo)' on a null object reference
       at io.flutter.plugin.editing.TextInputPlugin.createInputConnection(TextInputPlugin.java:41)
       at io.flutter.view.FlutterView.onCreateInputConnection(FlutterView.java:2)
       at android.view.inputmethod.InputMethodManager.startInputInner + 1237(InputMethodManager.java:1237)
       at android.view.inputmethod.InputMethodManager.onPostWindowFocus + 1529(InputMethodManager.java:1529)
       at android.view.ViewRootImpl$ViewRootHandler.handleMessage + 4105(ViewRootImpl.java:4105)
       at android.os.Handler.dispatchMessage + 106(Handler.java:106)
       at android.os.Looper.loop + 176(Looper.java:176)
       at android.app.ActivityThread.main + 6656(ActivityThread.java:6656)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run + 547(RuntimeInit.java:547)
       at com.android.internal.os.ZygoteInit.main + 873(ZygoteInit.java:873)

**Comment by huixisheng** [comment]
the same issue

**Comment by wingohtcom** [comment]
@amirh this happens when a webview containing a page that accepts input is disposed, and the app is minimized and then restore again.

Seems to be that the webview was disposed but flutter is still trying to establish a input connection?

**Comment by muhammadsaddamnur** [comment]
the same issue

**Comment by thedejifab** [comment]
> @amirh this happens when a webview containing a page that accepts input is disposed, and the app is minimized and then restore again.
> 
> Seems to be that the webview was disposed but flutter is still trying to establish a input connection?

@kingidee see this.

@wingohtcom found a fix yet?

**Comment by jyp334** [comment]
I can only say that there is no problem with dev version 1.7.11

**Comment by thedejifab** [comment]
> I can only say that there is no problem with dev version 1.7.11

@kingidee

**Comment by thedejifab** [comment]
I'm on `master` channel, version 1.9.1-pre.86 and it doesn't occur. How soon will changes be made to the `stable` channel?

**Comment by lzxb** [comment]
I also care.

**Comment by pythoneer** [comment]
Having the same with an app in production. My App has the functionality that, when a user sends my App into the background (e.g. using the system task switcher to view another app) my App logs out the user and send them back to the login screen when my App is going to get back to the foreground. The crash happens when the user was on a page with a webview, switches to another app or turn off the device screen and reactivates my App. This sounds very much like that roughly the same things happen as OP described.   

```
flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, v1.7.8+hotfix.4, on Linux, locale en_US.utf8)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
[✓] Android Studio (version 3.4)
[!] IntelliJ IDEA Ultimate Edition (version 2019.2)
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
[!] IntelliJ IDEA Community Edition (version 2018.2)
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
[!] Connected device
    ! No devices available

! Doctor found issues in 3 categories.
```

```
java.lang.NullPointerException: 
  at io.flutter.plugin.editing.TextInputPlugin.createInputConnection (TextInputPlugin.java:195)
  at io.flutter.view.FlutterView.onCreateInputConnection (FlutterView.java:397)
  at android.view.inputmethod.InputMethodManager.startInputInner (InputMethodManager.java:1763)
  at android.view.inputmethod.InputMethodManager$H.handleMessage (InputMethodManager.java:596)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loop (Looper.java:214)
  at android.app.ActivityThread.main (ActivityThread.java:7077)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:493)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:964)
```

**Comment by NewITCorp** [comment]
I have the same problem

[√] Flutter (Channel stable, v1.7.8+hotfix.4, on Microsoft Windows [Version 10.0.17763.678], locale en-US)
    • Flutter version 1.7.8+hotfix.4 at C:\flutter
    • Framework revision 20e59316b8 (5 weeks ago), 2019-07-18 20:04:33 -0700
    • Engine revision fee001c93f
    • Dart version 2.4.0

[√] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at C:\Users\pondk\AppData\Local\Android\sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)
    • All Android licenses accepted.

[√] Android Studio (version 3.5)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 38.2.3
    • Dart plugin version 191.8423
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)

[√] Connected device (1 available)
    • Android SDK built for x86 • emulator-5554 • android-x86 • Android 9 (API 28) (emulator)

• No issues found!





E/AndroidRuntime(22645): FATAL EXCEPTION: main
E/AndroidRuntime(22645): Process: com.newit.gisbkk, PID: 22645
E/AndroidRuntime(22645): java.lang.NullPointerException: Attempt to invoke virtual method 'android.view.inputmethod.InputConnection android.view.View.onCreateInputConnection(android.view.inputmethod.EditorInfo)' on a null object reference
E/AndroidRuntime(22645): 	at io.flutter.plugin.editing.TextInputPlugin.createInputConnection(TextInputPlugin.java:195)
E/AndroidRuntime(22645): 	at io.flutter.view.FlutterView.onCreateInputConnection(FlutterView.java:397)
E/AndroidRuntime(22645): 	at android.view.inputmethod.InputMethodManager.startInputInner(InputMethodManager.java:1290)
E/AndroidRuntime(22645): 	at android.view.inputmethod.InputMethodManager.onPostWindowFocus(InputMethodManager.java:1574)
E/AndroidRuntime(22645): 	at android.view.ViewRootImpl.handleWindowFocusChanged(ViewRootImpl.java:2593)
E/AndroidRuntime(22645): 	at android.view.ViewRootImpl.access$900(ViewRootImpl.java:132)
E/AndroidRuntime(22645): 	at android.view.ViewRootImpl$ViewRootHandler.handleMessage(ViewRootImpl.java:4175)
E/AndroidRuntime(22645): 	at android.os.Handler.dispatchMessage(Handler.java:106)
E/AndroidRuntime(22645): 	at android.os.Looper.loop(Looper.java:193)
E/AndroidRuntime(22645): 	at android.app.ActivityThread.main(ActivityThread.java:6669)
E/AndroidRuntime(22645): 	at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime(22645): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
E/AndroidRuntime(22645): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
I/Process (22645): Sending signal. PID: 22645 SIG: 9

**Comment by mklim** [comment]
It looks like this was fixed by flutter/engine#9727, which rolled into flutter at flutter/flutter#36000, or `v1.8.3`.

If anybody sees this happening when using flutter `v1.8.3` or later, please comment with `flutter doctor` and reproduction steps, if you have them. Thank you!
